tenable/io/v3/base/schema/explore/utils.py

```diff
@@ -0,0 +1,35 @@
+'''
+Utils for explore schema
+'''
+
+from typing import Dict, List, Tuple
+
+from tenable.errors import RequestValidationError
+from tenable.io.v3.base.schema.explore.search import SortSchema
+
+
+def generate_sort_data(
+        data: List, is_sort_with_prop: bool = True
+) -> List[Dict]:
+    schema = SortSchema()
+    sort_data = []
+    if isinstance(data, List):
+        for tup in data:
+            if isinstance(tup, Tuple) and len(tup) == 2:
+                property = tup[0]
+                order = tup[1]
+                data_validate = dict(property=property, order=order)
+                data_validate = schema.load(data_validate)
+                if is_sort_with_prop:
+                    sort_data.append(schema.dump(data_validate))
+                else:
+                    sort_data.append({property: order})
+            else:
+                raise RequestValidationError(
+                    'Sort', '[("field_name_1", "asc"),'
+                            ' ("field_name_2", "desc")]'
+                )
+    else:
+        sort_data = None
+
+    return sort_data
```

Comment:
This should be a pre_load method as part of the SortSchema instead.

```python
class SortSchema(Schema):
    @pre_load
    def tuple_conversion(self, data, **kwargs):
        if is instance(data, tuple):
            data = {'property': data[0], 'order': data[1]}
        return data
```

Refer to https://github.com/dhvanilshah-tenable/pyTenable/blob/8991bcdf5703ee4a514468ef1aa1119df38693ae/tenable/io/v3/base/schema/explore/filters.py as a good example of this.